src/parser.ts

```diff
@@ -103,16 +98,43 @@ class V1Validator {
     }
   }
 
-  recordError(msg: string) {
-    this.errors.push(
-      new ValidationError(msg, this.getLocation())
+  /**
+   * Validates the root names of the doc.
+   * We just do this by hand as it's tricky to do recursively
+   */
+  validateRootNames() {
+    const exports = this.doc.exports || {}
+    for (const n in exports) {
+      this.validateIdentifier(n, ['exports', n])
+    }
+    const imports = this.doc.imports || {}
+    for (const n in imports) {
+      this.validateIdentifier(n, ['imports', n])
+    }
+    const schemas = this.doc.components?.schemas || {}
+    for (const n in schemas) {
+      this.validateIdentifier(n, ['components', 'schemas', n])
```

Comment:
I'm kind of considering this alternative interface for this pattern:

```typescript
      this.validateIdentifier(n, `components/schemas/${n}`)
```

I think this is understandable for now though